pkg/mflag/flag.go

```diff
@@ -778,8 +840,8 @@ func Var(value Value, names []string, usage string) {
 // returns the error.
 func (f *FlagSet) failf(format string, a ...interface{}) error {
 	err := fmt.Errorf(format, a...)
-	fmt.Fprintln(f.out(), err)
-	f.usage()
+	fmt.Fprintln(f.Out(), err)
+	fmt.Fprintf(f.Out(), "See 'docker %s --help'.\n", f.name)
```

Comment:
Why?   Docker run spews tuns of data onto the screen can causes the top lines to roll off with the valuable error.

If I want usage I should ask for it.

Standard linux tools like coreutils work this way.

ls --fobar
ls: unrecognized option '--fobar'
Try 'ls --help' for more information.